src/Orleans.Runtime/MembershipService/ILegacyMembershipConfigurator.cs

```diff
@@ -56,7 +56,15 @@ private class LegacyGrainBasedMembershipConfigurator : ILegacyMembershipConfigur
         {
             public void ConfigureServices(GlobalConfiguration configuration, IServiceCollection services)
             {
-                services.UseGrainBasedMembership();
+                services.UseGrainBasedMembership(options => CopyGlobalGrainBasedMembershipOptions(configuration, options));
+            }
+
+            private static void CopyGlobalGrainBasedMembershipOptions(GlobalConfiguration configuration, GrainBasedMembershipOptions options)
+            {
+                if (configuration.SeedNodes?.Count > 0)
+                {
+                    options.SeedNode = configuration.SeedNodes?.FirstOrDefault()?.ToGatewayUri();
```

Comment:
ToGatewayUri here seems a bit odd to me. I think the `SeedNode` property should be a `SiloAddress` instead of a `Uri`

Reply:
yeah, I'm going back and forth, the thing is that it is not serializable as a string, so you cannot set it from external config...

Reply:
can we somehow fix that? Implicit constructor?

Reply:
Given we will never know the `generation`, we could use IPEndpoint

Reply:
with `IPEndpoint` we can't use implicit constructor for sure though. With SiloAddress it *might* work, but it's true that there is no generation info.
I could use a `string` as well, but it's less clear what is the expected format